sigs/addons/MIGRATION_RFC_TEMPLATE.md

```diff
@@ -0,0 +1,41 @@
+# Migrate XXXXX from TensorFlow Addons to TensorFlow Core
+
+| Status      | Proposed (Waiting for Sponsor)                                                                                           |
+| :---------- | :------------------------------------------------------------------------------------------------- |
+| **RFC #**   | TBD after PR |                                       |
+| **Authors** | XXXXXXXXXX   |
+| **Sponsor** | XXXXXXXXXX   |
+| **Updated** | YYYY-MM-DD   |
+| **Sponsorship Deadline** | YYYY-MM-DD (45 Days after submission)  |
+
+## Rationale for Migration
+* What are the use cases for the addon?
+* OSS usage, H5 Index, etc.
+
+## Historical Information
+* Have there been signifiant issues reported to Addons that need to be adressed?
+* When was it implemented in Addons?
+
+## Implementation Details
+* Link to implementation in Addons:
+* Does this include custom-op kernels?
+    * Are they CPU/GPU/TPU compatible?
+* What is the pytest coverage of the addon?
+
+## Changes to Implementation (If Needed)
+```
+Code snippet(s) showing proposed code changes
+```
+* Discussion on the rationale for changes
+
+## Transition Plan
+* Proposed landing place in tf-core
+* Deprecation plan for Addons
+    * Will we be able to alias the core implementation? (parameters must be exact match)
```

Comment:
I think that the upstreamed version could, under a review request, acquire a parameter superset. But if the additional paramas are optional I think that it is still ok if we have the same output with the parameter subset.

Reply:
We shouldn't alias it, but use the tf-core function in the body of our implementation. Otherwise we can't add a deprecation warning somewhere. As such it's ok if the parameters differs as long as it offers the same features/functionalities.

Reply:
I'd like to leave the core implementation the freedom to drop parameters that are not frequently used, and rename other parameters which might have better names.

Reply:
Thanks, yeah even if it were a parameter match we'd still want to call the core function inside of our call with deprecation as mentioned. Updated the discussion points to keep them more general.